index.js

```diff
@@ -1687,7 +1742,62 @@
   //. //
   //. //   The value at position 1 is not a member of ‘{ x :: FiniteNumber, y :: FiniteNumber }’.
   //. ```
-  function RecordType(fields) {
+  //.
+  //. ##### Extending record types
+  //.
+  //. By using the `parent` argument of the `RecordType` constructor, one can
+  //. define record types which must contain all the same properties of
+  //. another. For example, one could define `Point3D` as a refinement of
+  //. `Point` from the previous example:
+  //.
+  //. ```javascript
+  //. //    Point3D :: Type
+  //. const Point3D = $.RecordType(Point, {z: $.FiniteNumber});
+  //.
+  //. //    p1 :: Point
+  //. const p1 = {x: 0, y: 0};
+  //.
+  //. //    p2 :: Point, Point3D
+  //. const p2 = {x: 3, y: 4, z: 5};
+  //.
+  //. Point.validate(p1);    // => Right({x: 0, y: 0})
+  //. Point3D.validate(p1);  // => Left({propPath: ['z'], value: undefined})
+  //.
+  //. Point.validate(p2);    // => Right({x: 3, y: 4, z: 5})
+  //. Point3D.validate(p2);  // => Right({x: 3, y: 4, z: 5})
+  //.
+  //. dist(p1, p2);
+  //. // => 5
+  //. ```
+  //.
+  //. The `p2` value is a member of `Point3D` as well as `Point`, making it a
+  //. valid argument to `dist`. By default record types allow additional
+  //. properties (which is what allowed `p2` to be a member of `Point`). If one
+  //. wishes for a record type to allow for no additional properties, it can be
+  //. refined as such:
+  //.
+  //. ```javascript
+  //. //    strict :: Type -> Type
+  //. const strict = t => $.NullaryType(
+  //.   'Strict' + t.name,
+  //.   '',
+  //.   t,
+  //.   x => Object.keys(x).length === t.keys.length
+  //. );
```

Comment:
I have a much stronger argument:

```js
> var User = $.RecordType($.Any, {name: $.NonEmpty($.String)})

> User.validate({name: ''})
Left({ value: '', propPath: [ 'name' ] })

> strict(User).validate({name: ''});
Left({ value: '', propPath: [ 'name' ] })

> Strict(User)
Left({ value: { name: '' }, propPath: [] })
```

Or in words: Your `UnaryType` relies on `$.test`, and the purpose of parent types was to fix `validate` for those types that relied on `test`-ing another type. Actually, the purpose of this PR was to fix #128, and in particular, to address my comment there: https://github.com/sanctuary-js/sanctuary-def/issues/128#issuecomment-287705497